skywalking-sniffer/skywalking-sdk-plugin/motan-plugin/src/main/java/com/a/eye/skywalking/plugin/motan/ProviderInterceptor.java

```diff
@@ -0,0 +1,92 @@
+package com.a.eye.skywalking.plugin.motan;
+
+import com.a.eye.skywalking.api.context.ContextCarrier;
+import com.a.eye.skywalking.api.context.ContextManager;
+import com.a.eye.skywalking.api.plugin.interceptor.EnhancedClassInstanceContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.ConstructorInvokeContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceMethodInvokeContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.MethodInterceptResult;
+import com.a.eye.skywalking.trace.Span;
+import com.a.eye.skywalking.trace.tag.Tags;
+import com.weibo.api.motan.rpc.Request;
+import com.weibo.api.motan.rpc.Response;
+import com.weibo.api.motan.rpc.URL;
+
+/**
+ * Current trace segment will ref the trace segment from previous level if the serialized context data that fetch
+ * from {@link Request#getAttachments()} is not null.
+ *
+ * {@link ProviderInterceptor} intercept all constructor of {@link com.weibo.api.motan.rpc.AbstractProvider} for record
+ * the request url from consumer side.
+ *
+ * @author zhangxin
+ */
+public class ProviderInterceptor implements InstanceConstructorInterceptor, InstanceMethodsAroundInterceptor {
+
+    /**
+     * The key name of request url that stored in {@link EnhancedClassInstanceContext#context}
+     */
+    private static final String KEY_NAME_OF_REQUEST_URL = "REQUEST_URL";
+
+    /**
+     * The key name that the serialized context data stored in {@link Request#getAttachments()}
```

Comment:
The {@link Request#getAttachments()} key. It maps to the serialized  {@link ContextCarrier}.